packages/desktop-gui/src/lib/loader.jsx

```diff
@@ -0,0 +1,15 @@
+import cs from 'classnames'
+import React from 'react'
+
+const Loader = ({ children, className = '', fullscreen = false }) => (
+  <span className={cs('loader', className,
+    { 'loader-fullscreen': fullscreen }
```

Comment:
It feels like part of the responsibly of a loader is to provide a scrim over the appropriate content. This can either be the entire viewport (fullscreen) or any element (non-fullscreen). However, in both cases, it prevents the user from taking action until after the loader has completed. This seems like the intended purpose of a loading screen/loader.

I am not suggesting to bake into `Loader` a scrim. The concept of a scrim stands on its own. However, we can leverage a Scrim (ui-block, etc.) and compose it within `Loader`.

Specifically, my suggestions are:
- compose Loader with a scrim (or ui-block; doesn't have be named Scrim)
- full screen loaders show a scrim the entire viewport
- non-full screen loaders show a scrim over their parent consumer
    - if we simply do not need a non-full screen loader, then consider naming this `FullScreenLoader` instead. If we come back in the future with a need for a non-fullscreen version then it can simply compose the `FullScreenLoader` within it.

Reply:
Note: I would defer the non-fullscreen loader if possible; as it is more complicated in terms of implementation.

Edit: It appears like there is at least [one usage](https://github.com/cypress-io/cypress/pull/3268/files#diff-2f9b8415c2aa516569e9ba1e934f198cR60) of a non-fullscreen loader.